src/core/imports/cuda.cpp

```diff
@@ -37,6 +37,12 @@ void InitializeCUDA(int argc, char* argv[])
             env = std::getenv("MV2_COMM_WORLD_LOCAL_RANK");
         if (!env)
             env = std::getenv("OMPI_COMM_WORLD_LOCAL_RANK");
+        if (!env)
+            // This returns a list if more than one gpu is present in the
+            // resource set, for example:
+            // CUDA_VISIBLE_DEVICES=0,1,2,3
+            // std::atoi picks the first one
+            env = std::getenv("CUDA_VISIBLE_DEVICES");
```

Comment:
Am I correct then that this assumes the on-node GPUs are reordered such that each process sees a different GPU as "device 0"?

Reply:
Yes

```
[mcneish1@sierra4360:spack-dev]$ bsub -nnodes 4 -W10 -Is -G guests jsrun -n 16 -g 1 env | grep CUDA_VISIBLE_DEVICES
<<Waiting for dispatch ...>>
<<Starting on sierra4369>>
CUDA_VISIBLE_DEVICES=0
CUDA_VISIBLE_DEVICES=1
CUDA_VISIBLE_DEVICES=2
CUDA_VISIBLE_DEVICES=1
CUDA_VISIBLE_DEVICES=0
CUDA_VISIBLE_DEVICES=0
CUDA_VISIBLE_DEVICES=3
CUDA_VISIBLE_DEVICES=2
CUDA_VISIBLE_DEVICES=1
CUDA_VISIBLE_DEVICES=3
CUDA_VISIBLE_DEVICES=2
CUDA_VISIBLE_DEVICES=3
CUDA_VISIBLE_DEVICES=1
CUDA_VISIBLE_DEVICES=0
CUDA_VISIBLE_DEVICES=2
CUDA_VISIBLE_DEVICES=3
```

Reply:
Hi face, meet palm. Ok. I think this is fine then.